### DISCUSSION TITLE
How to give user the permission to List or Retrieve from within TS file programatically

### DISCUSSION THREAD

**Discussion by sken78** [serenity-is/Serenity#6400] [Help] [answered]
I would like to retrieve current user Roles inside a TS file. But that user has no permission to list the roles. How can I give him the permission to do this and than remove it in the next line of code.

If I could always use the admin user for such tasks, this kind of issues will be solved. 

Thanks

**Comment by wezmag** [comment]
To retrieve current user's roles, you can use `Q.getRemoteData('UserData')` in your script.
In `Permissions`, you can find permission keys begin with `Role:`.
That's the current user's roles.
``` json
{
    "Username": "wesley.huang@contoso.com", //not my real email
    "DisplayName": "Wesley Huang",
    "IsAdmin": false,
    "Permissions": {
        "Role:Customer": true,
        "Role:PremiumCustomer": true
    }
}
```

If you can not find the role, please make sure your role has "Role Key".

**Comment by wirble** [comment, marked as answer]
Another way is doing this in the xxxGrid.ts file probably in the constructor or on before load.  Either case, it probably not the secure way of doing it if that is a concern.  

```
if (!Q.Authorization.hasPermission("Admin"))
                if (!Q.Authorization.hasPermission("Records:Editable"))
                    this.set_readOnly(true);
```

I believe you can either use the name of the Role or the permission keys.
